### ISSUE TITLE
Allow more generic R methods

### ISSUE THREAD

**Issue by qykong** [mlverse/torch#164]
Hi, thanks for the effort in developing this package 👍 . I've been following the development of the package and am quite excited about the idea. After quickly trying the tutorials, I notice that `operators.R` currently only implements the S3 methods for some operations. I wonder if there will be any plans to implement more  math generic methods such as `log`, `sum`, `mean`, etc. In my case, this would allow me to reuse my existing R code base and apply autograd easily. If this sounds interesting but isn't the current focus, I'm keen to contribute 😃

**Comment by dfalbel** [comment]
HI @qykong 

Thanks for your message!

It's not on the short term plan to implement other generics. But yes! I think it's a great idea, I would happily review a PR implementing it.

My only concern is  `sum`: we should be able to support other arguments like `dim`, `keepdim`, but that should be fine with most generics.

**Comment by qykong** [comment]
Hi @dfalbel , thanks for your reply. I'm working on this now and I agree with your concern about functions like `sum`. I'll include the extra arguments as well.

While most generics are straight-forward, I notice some other discrepancies of some functions implemented in R and pytroch.  For example, `min(c(1,2,3), c(2,3,4))` in R gives a single number while `torch.min(torch.tensor([1,2,3]), torch.tensor([2,3,4]))` returns a tensor `tensor([1,2,3])` in pytorch. In such cases, I think it might be better to follow the default R behaviour (i.e., returning a single value as in the example). What do you think?
